File: frontend/src/dto/DashboardDTO.js
// DTOs para el Dashboard
export class DashboardDTO {
  constructor() {
    this.empresa = ""
    this.vencimientosProximos = []
    this.indicadores = new IndicadoresDTO()
    this.alertasInconsistencias = []
  }
}

export class IndicadoresDTO {
  constructor() {
    this.posicionIva = 0
    this.ivaDebito = 0
    this.ivaCredito = 0
    this.saldoIIBB = 0
    this.saldoGanancias = 0
  }
}

export class VencimientoDTO {
  constructor() {
    this.id = null
    this.descripcion = ""
    this.fechaVencimiento = null
    this.montoEstimado = 0
    this.estado = ""
    this.tipoImpuesto = ""
    this.diasRestantes = 0
  }
}

File: backend/src/main/java/com/contable/dto/DashboardDTO.java
package com.contable.dto;

import lombok.Data;
import java.math.BigDecimal;
import java.util.List;

@Data
public class DashboardDTO {
    private String empresa;
    private List<VencimientoDTO> vencimientosProximos;
    private IndicadoresDTO indicadores;
    private List<String> alertasInconsistencias;
    
    @Data
    public static class IndicadoresDTO {
        private BigDecimal posicionIva = BigDecimal.ZERO;
        private BigDecimal ivaDebito = BigDecimal.ZERO;
        private BigDecimal ivaCredito = BigDecimal.ZERO;
        private BigDecimal saldoIIBB = BigDecimal.ZERO;
        private BigDecimal saldoGanancias = BigDecimal.ZERO;
    }
}
